src/components/Router/StepComponentMap.js

```diff
@@ -0,0 +1,53 @@
+import { h, Component } from 'preact'
+import Welcome from '../Welcome'
+import Select from '../Select'
+import {FrontDocumentCapture, BackDocumentCapture, FaceCapture} from '../Capture'
+import Complete from '../Complete'
+
+export const steps = (stepOptions, documentType) => {
+  const mapOptions = stepOption => stepOptionToStep(stepOption, documentType)
+  return shallowFlatten(stepOptions.map(mapOptions))
+}
+
+export const components = (stepList, componentOptions) => {
+  const mapSteps = step => stepToComponents(step, componentOptions)
+  return stepList.map(mapSteps)
+}
+
+const stepOptionToStep = (stepOption, documentType) => {
+  const step = formatStep(stepOption)
+  return step.type === 'document' ? documentStepsList(step.options, documentType) : step
+}
+
+const documentStepsList = (options, documentType) => {
+  const steps = [{type: 'documentSelect', ...options},
+                 {type: 'documentFront', ...options}]
+  if (['driving_licence', 'national_identity_card'].includes(documentType)) {
```

Comment:
is there now two levels of indirection? configuration steps -> intermediate steps -> components?